cli.js

```diff
@@ -170,11 +166,73 @@ Usage: $0 <command> [options]` // eslint-disable-line comma-dangle
   .command('list [options]', 'List all deployed functions', (yargs0) => {
     yargs0
       .usage('Usage: $0 list [options]')
-      .option('json', { describe: 'Output as JSON' })
+      .option('json', {
+        describe: 'Output as JSON',
+        type: 'boolean',
+      })
       .strict();
   }, async (argv) => {
     await handleCommand(argv, listHandler);
   })
+  .command('trigger', 'Manages triggers', yargs0 => yargs0
+      .demandCommand()
+      .command('setup', 'Creates a new trigger or updates an existing one', (yargs1) => {
+        yargs1
+          .usage('Usage: $0 trigger setup <trigger name> [options]')
+          .positional('trigger name', {
+            describe: 'Trigger name',
+            type: 'string',
+          })
+          .option('source-uri', {
+            demandOption: true,
+            describe: 'Redis-stream uri',
+            type: 'string',
+          })
+          .option('target', {
+            demandOption: true,
+            describe: 'Function to run for each item in the stream',
+            type: 'string',
+          })
+          .option('error-handler', {
+            describe: 'Function to run when an error occurs',
+            type: 'string',
+          })
+          .demandCommand(1)
+          .strict()
+          .example(
+`  // Creates trigger 'my-trigger' that will invoke 'my-func' for every entry in 'my-stream'
+  bn trigger setup my-trigger --source-uri "redis://:redispass@redis.foo.com/0#my-stream" --target my-func
+
+  // Same this but secured
+  bn trigger setup my-trigger --source-uri "redis+tls://:redispass@redis.foo.com/0#my-stream" --target my-func`,
```

Comment:
Need an explanation, like so:
```
PROTOCOL://:PASSWORD@HOST       :PORT/DB#STREAM_NAME
redis   ://:secret  @example.com:6379/0 #numbers
```

Where PORT, PASSWORD and DB are all optional.

Reply:
done